plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/proxy/NugetProxyGalleryFacet.java

```
@@ -179,27 +195,30 @@ private URI nugetQuery(final String path, final Map<String, String> queryParams)
    * using the supplied {@link RemoteCallFactory}.
    *
-   * @return the package count from the remote repo
+   * @return successfully returned counts, which might have fewer entries than repositories.size (or none at all)
    */
-  private int passQueryToRemoteRepo(final URI path, final RemoteCallFactory remoteCall)
+  private List<Integer> passQueryToRemoteRepos(final URI path, final Iterable<Repository> repositories,
+                                               final RemoteCallFactory remoteCall)
   {
-    final Repository repo = getRepository();
+    final List<Integer> counts = new ArrayList<>();
 
-    try {
-      // TODO: Determine if we should talk to the remote based on its status
+    for (Repository repo : repositories) {
+      try {
+        // TODO: Determine if we should talk to the remote based on its status
 
-      final QueryCacheKey key = new QueryCacheKey(repo.getName(), path);
-      final Integer cachedCount = cache.get(key, remoteCall.build(repo, path));
-      return cachedCount;
-    }
-    catch (ExecutionException | UncheckedExecutionException e) {
-      log.warn("{} attempting to contact proxied repository {}.", e.getCause().getClass().getSimpleName(),
-          repo.getName());
-    }
-    catch (Exception e) {
-      log.warn("{} attempting to contact proxied repository {}.", e.getClass().getSimpleName(), repo.getName());
+        final QueryCacheKey key = new QueryCacheKey(repo.getName(), path);
+        final Integer cachedCount = cache.get(key, remoteCall.build(repo, path));
+        counts.add(cachedCount);
+      }
+      catch (ExecutionException | UncheckedExecutionException e) {
+        log.warn("{} attempting to contact proxied repository {}.", e.getCause().getClass().getSimpleName(),
+            repo.getName());
+      }
+      catch (Exception e) {
+        log.warn("{} attempting to contact proxied repository {}.", e.getClass().getSimpleName(), repo.getName());
+      }
```

both of these can eat exception detail, was this on purpose?